argilla-server/tests/unit/contexts/hub/test_hub_dataset_exporter.py

```diff
@@ -0,0 +1,788 @@
+#  Copyright 2021-present, the Recognai S.L. team.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+import os
+import pytest
+
+from PIL import Image
+from uuid import uuid4
+from typing import Generator
+from huggingface_hub import HfApi
+from datasets import load_dataset, get_dataset_config_names, get_dataset_split_names
+
+from argilla_server.contexts.hub import HubDatasetExporter
+from argilla_server.enums import DatasetStatus, FieldType, QuestionType, ResponseStatus, MetadataPropertyType
+
+from tests.factories import (
+    DatasetSyncFactory,
+    FieldSyncFactory,
+    QuestionSyncFactory,
+    RecordSyncFactory,
+    ResponseSyncFactory,
+    AnnotatorSyncFactory,
+    MetadataPropertySyncFactory,
+    VectorSettingsSyncFactory,
+    VectorSyncFactory,
+)
+
+HF_ORGANIZATION = "argilla-internal-testing"
+HF_TOKEN = os.environ.get("HF_TOKEN_ARGILLA_INTERNAL_TESTING")
+
+IMAGE_URL = "https://argilla.io/brand-assets/argilla/argilla-logo-color-black.png"
+IMAGE_DATA_URL = "data:image/gif;base64,R0lGODlhAQABAAAAACH5BAEKAAEALAAAAAABAAEAAAICTAEAOw=="
+
+
+@pytest.fixture
+def hf_api() -> HfApi:
+    return HfApi(token=HF_TOKEN)
+
+
+@pytest.fixture
+def hf_dataset_name(hf_api: HfApi) -> Generator[str, None, None]:
+    hf_dataset_name = f"{HF_ORGANIZATION}/argilla-server-dataset-test-{uuid4()}"
+
+    yield hf_dataset_name
+
+    hf_api.delete_repo(hf_dataset_name, repo_type="dataset", missing_ok=True)
+
```

Comment:
Why not prepare the test session here instead of defining a `async_client` argument, which can be hard to understand?

```suggestion
from argilla_server.contexts import hub 

@pytest.fixture
def test_sync_db():
   session = SyncTestSession()
   
   def override_get_sync_db():
        yield session

   mocker.patch.object(hub, "get_sync_db", override_get_sync_db)

   yield session
```

Reply:
Ok, let's move it to the test.